File: Frontend/src/modules/admin/feature/stockMovement/types/stock-movement.type.ts
export type StockMovementType =
  | "RECEIPT_IN"
  | "OPENING_IN"
  | "SALE_OUT"
  | "RETURN_IN"
  | "ADJUST_IN"
  | "ADJUST_OUT"
  | "DISPOSAL_OUT";

export interface StockMovementListResponse {
  id: number;
  createdAt: string;

  inventoryLotId: number;
  productVariantId: number;

  sku: string;
  productName: string;
  lotCode: string;

  movementType: StockMovementType | null;
  movementTypeLabel: string | null;

  quantityChange: number;
  quantityBefore: number;
  quantityAfter: number;

  referenceType: string | null;
  referenceId: number | null;
  referenceLineId: number | null;

  reason: string | null;

  createdById: number;
  createdByName: string;
}

export interface StockMovementDetailResponse
  extends StockMovementListResponse {}

export interface StockMovementListParams {
  keyword?: string;
  inventoryLotId?: number | null;
  movementType?: StockMovementType | "";
  createdBy?: number | null;
  referenceType?: string;
  referenceId?: number | null;
  fromDate?: string;
  toDate?: string;
  page?: number;
  size?: number;
}

export interface PageMeta {
  size: number;
  number: number;
  totalElements: number;
  totalPages: number;
}

export interface PageResponse<T> {
  content: T[];
  page?: PageMeta;
  size?: number;
  number?: number;
  totalElements?: number;
  totalPages?: number;
}

export const STOCK_MOVEMENT_TYPE_OPTIONS: Array<{
  value: StockMovementType;
  label: string;
}> = [
  { value: "RECEIPT_IN", label: "Nhập kho" },
  { value: "OPENING_IN", label: "Tồn đầu kỳ" },
  { value: "SALE_OUT", label: "Xuất bán" },
  { value: "RETURN_IN", label: "Nhập trả hàng" },
  { value: "ADJUST_IN", label: "Điều chỉnh tăng" },
  { value: "ADJUST_OUT", label: "Điều chỉnh giảm" },
  { value: "DISPOSAL_OUT", label: "Xuất hủy" },
];

export const REFERENCE_TYPE_OPTIONS = [
  { value: "GOODS_RECEIPT", label: "Phiếu nhập kho" },
  { value: "ORDER", label: "Đơn hàng" },
  { value: "RETURN", label: "Phiếu trả hàng" },
  { value: "STOCK_ADJUSTMENT", label: "Điều chỉnh kho" },
];

File: Backend/src/main/java/org/example/datn_sd69/enums/StockMovementType.java
package org.example.datn_sd69.enums;

import lombok.Getter;

@Getter
public enum StockMovementType {

    RECEIPT_IN((byte) 1, "Nhập kho"),

    OPENING_IN((byte) 2, "Tồn đầu kỳ"),

    SALE_OUT((byte) 3, "Xuất bán"),

    RETURN_IN((byte) 4, "Nhập trả hàng"),

    ADJUST_IN((byte) 5, "Điều chỉnh tăng"),

    ADJUST_OUT((byte) 6, "Điều chỉnh giảm"),

    DISPOSAL_OUT((byte) 7, "Xuất hủy");

    private final Byte code;

    private final String label;

    StockMovementType(
            Byte code,
            String label
    ) {
        this.code = code;
        this.label = label;
    }

    public static StockMovementType fromCode(Byte code) {

        if (code == null) {
            return null;
        }

        for (StockMovementType type : values()) {

            if (type.code.equals(code)) {
                return type;
            }
        }

        return null;
    }
}